### ISSUE TITLE
`bw test -p` broken?

### ISSUE THREAD

**Issue by vain** [bundlewrap/bundlewrap#656]
```python
@metadata_reactor.provides(
    'foo/bar/baz',
)
def react(metadata):
    return {
        'foo': {
            'bar': {
                'baz is not included in the response',
            },
        },
    }
```
```
$ bw test -p node-1
✓ No reactors violated their declared keys
$ 
```
Am I misunderstanding or is it borked?

**Comment by vain** [comment]
Ah. It’s a grey area.

`.provides()` only specifies a “limit”. You cannot return keys that you didn’t declare. But it doesn’t specify that you *have* to return all those keys. I’m afraid that is a valid use case, and we make use of this (a lot).

In the example above, the reactor returned something which is a “subpath” of `foo/bar/baz`: We returned `foo/bar`. This is … valid? Or is it?

The most “conservative” fix would be: For each declared subpath, check that it really is a dict. In this case, `foo/bar` *must* be a dict (otherwise `foo/bar/baz` couldn’t exist).

**Comment by trehn** [comment]
fixed in 6840f8775ed8c612b2dd31bc73afcf33ee2ac2b8
